### PR TITLE
Update comfy example to emphasize how to iterate & avoid redownloading weights

### PR DESCRIPTION
**Pull request by kramstrom**
~Basically adding parts of [this gist](https://gist.github.com/kramstrom/a3784d8246948db647e7e66df980d090) as further improvements to our comfy example. It does mainly two things:~
~- store model weights on a volume so that they can be re-used across different apps and allows you to change your image however you want it without having to re-download everything~
~- it changes the model downloading to use huggingface helpers instead which allows you to pass in a huggingface token and download protected models~

~Not sure if these fit best as inline code here or if the gist should be its own example / guide / blog post or scrapped completely and just remain as a gist.~

Updated the example instead to split up the image building steps into multiple & explain each step more thoroughly and how users should add new models and custom nodes without having to rebuild the whole image (which is why users want to store them on volumes in the first place)

### Type of Change

- [ ] New example
- [x] Example updates (Bug fixes, new features, etc.)
- [ ] Other (changes to the codebase, but not to examples)

## Checklist

- [ ] Example is testable in synthetic monitoring system, or `lambda-test: false` is added to example frontmatter (`---`)
  - [ ] Example is tested by executing with `modal run` or an alternative `cmd` is provided in the example frontmatter (e.g. `cmd: ["modal", "deploy"]`)
  - [ ] Example is tested by running with no arguments or the `args` are provided in the example frontmatter (e.g. `args: ["--prompt", "Formula for room temperature superconductor:"]`
- [ ] Example is documented with comments throughout, in a [_Literate Programming_](https://en.wikipedia.org/wiki/Literate_programming) style.
- [ ] Example does _not_ require third-party dependencies to be installed locally
- [ ] Example pins its dependencies
  - [ ] Example pins container images to a stable tag, not a dynamic tag like `latest`
  - [ ] Example specifies a `python_version` for the base image, if it is used
  - [ ] Example pins all dependencies to at least minor version, `~=x.y.z` or `==x.y`
  - [ ] Example dependencies with `version < 1` are pinned to patch version, `==0.y.z`

## Outside contributors

You're great! Thanks for your contribution.

### CODE REVIEW

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-cc6c646.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-5b2f894.modal.run

**Review by charlesfrye — CHANGES_REQUESTED** [review]
Discussed offline -- we don't want to add this as a block of untested code at the end. Kasper is rewriting to merge this with the rest of the material.

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-12cfa19.modal.run

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-2e5b652.modal.run

File: 06_gpu_and_ml/comfyui/comfyapp.py [hunk 1]
```diff
@@ -54,33 +52,97 @@
...
 
+# ### Building up the environment
+#
+# ComfyUI setups can be complex, with a lot of custom nodes and models to manage.
+# We'll use [`comfy-cli`](https://github.com/Comfy-Org/comfy-cli) to manage the installation of ComfyUI, its dependencies, models, and custom nodes.
+#
+# We start from a base image and specify all of our dependencies.
+# We'll call out the interesting ones as they come up below.
+#
+# Note that these dependencies are not installed locally
```
Comment (charlesfrye) [inline 06_gpu_and_ml/comfyui/comfyapp.py:63]:
```suggestion
# Note that these dependencies are not installed locally.
```

File: 06_gpu_and_ml/comfyui/comfyapp.py [hunk 2]
```diff
@@ -54,33 +52,97 @@
...
+# they are only installed once in the remote environment where our app runs,
```
Comment (charlesfrye) [inline 06_gpu_and_ml/comfyui/comfyapp.py:64]:
```suggestion
# They are only installed in the remote environment where our app runs.
```

File: 06_gpu_and_ml/comfyui/comfyapp.py [hunk 3]
```diff
@@ -54,33 +52,97 @@
...
+# on subsequent runs, the image will be cached and reused.
```
Comment (charlesfrye) [inline 06_gpu_and_ml/comfyui/comfyapp.py:65]:
```suggestion
# This happens the first time. On subsequent runs, the cached image will be reused.
```

File: 06_gpu_and_ml/comfyui/comfyapp.py [hunk 4]
```diff
@@ -54,33 +52,97 @@
...
+        "comfy --skip-prompt model download --url https://huggingface.co/black-forest-labs/FLUX.1-schnell/resolve/main/flux1-schnell.safetensors --relative-path models/unet"
     )
-    .run_commands(  # download a custom node
+    # Add .run_commands(...) calls for any other models you want to download
+)
+
+# #### Downloading custom nodes
+#
+# We'll download custom nodes using `comfy-cli` too.
+# (alternatively, you can install them by cloning the git repositories to your `/root/comfy/ComfyUI/custom_nodes`
```
Comment (charlesfrye) [inline 06_gpu_and_ml/comfyui/comfyapp.py:109]:
```suggestion
# Alternatively, you can install them by cloning the git repositories to your `/root/comfy/ComfyUI/custom_nodes`
```

File: 06_gpu_and_ml/comfyui/comfyapp.py [hunk 5]
```diff
@@ -54,33 +52,97 @@
...
+# directory and installing the required dependencies manually.)
```
Comment (charlesfrye) [inline 06_gpu_and_ml/comfyui/comfyapp.py:110]:
```suggestion
# directory and installing the required dependencies manually.
```

**Comment by charlesfrye** [comment]
🚀 The docs preview is ready! Check it out here:     https://modal-labs-examples--frontend-preview-8c3c5c7.modal.run

**Review by charlesfrye — APPROVED** [review]

**Comment by kramstrom** [comment]
> it looks like here then that we're not publishing the volume-based approach and just telling people to append dependencies to the image definition to leverage caching?

Yeah we opted to go for the current approach with appending dependencies, if people really want to use volumes though we still have the gist to show how it could be done

**Pull request merged**